### ISSUE TITLE
stat_cor() - Label position not consistent across plots

### ISSUE THREAD

**Issue by fdchevalier** [kassambara/ggpubr#248]
Dear Alboukadel,

First, thank you for developing and maintaining this very useful package.

I recently made a panel of two correlation plots (see example below). I positioned the correlation coefficient and p-values using character designation for `label.x.npc` and `label.y.nc`. While the position was as expected for each plot, I was surprised to see that they did not align. The figure does not look its best to me.

![panel_example](https://user-images.githubusercontent.com/27694997/77485274-bab69180-6dfa-11ea-90aa-df3b0f4d2d3e.png)

From my understanding looking at the plot, the label are positioned following the highest value in the dataset minus a proportional value. The highest value obviously varies between datasets. It may be better to use the max of the axis to position the label. This way, position will be consistent across plots. With basic plot, this can be obtained using `par("usr")[4]` but I don't know if such data can be accessible using ggplot object (I don't have any knowledge in manipulating gg objects). I looked at the code but I did not find anything obvious to modify.

Below is a minimum reproducible example that generates the figure.

Thank you for your help with this and hope that you and your family are going well in this difficult time.

Fred

```R
library("ggpubr")

# First data frame
a <- structure(list(ReplicateID = structure(1:42, .Label = c("A_1",
"A_2", "A_3", "A_4", "A_5", "A_H", "C_1", "C_2",
"C_3", "C_4", "C_5", "C_H", "B_1", "B_2",
"B_3", "B_4", "B_5", "B_H", "D_1", "D_2",
"D_3", "D_4", "D_5", "D_H", "E_1", "E_2",
"E_3", "E_4", "E_5", "E_H", "F_1", "F_2",
"F_3", "F_4", "F_5", "F_H", "G_1",
"G_2", "G_3", "G_4", "G_5", "G_H"
), class = "factor"), Population = structure(c(1L, 1L, 1L, 1L,
1L, 8L, 3L, 3L, 3L, 3L, 3L, 8L, 2L, 2L, 2L, 2L, 2L, 8L, 4L, 4L,
4L, 4L, 4L, 8L, 5L, 5L, 5L, 5L, 5L, 8L, 6L, 6L, 6L, 6L, 6L, 8L,
7L, 7L, 7L, 7L, 7L, 8L), .Label = c("A", "B", "C", "D",
"E", "F", "G", "H"), class = "factor"), V1 = c(0.0351330261596506,
0.085270637792105, 0.0355937581982821, 0.0226259451893908, 0.0214030482099555,
0.261743159540905, 0.0471092309127392, 0.0676759968796377, 0.0656883520809634,
0.0721555799021021, 0.0552352653849761, 0.0681070186185766, 0.0734105305241865,
0.0826089842807716, 0.0988981933073463, 0.036593694497667, 0.0406553242788484,
0.00845856485933266, 0.0297162025150854, 0.0421547841430283,
0.0224452743955855, 0.0266374857862149, 0.0258558284409483, 0.0522791197297753,
0.0916918146952398, 0.0441566072234197, 0.0459580463386767, 0.0786892082936663,
0.0576343040088014, 0.0608076780872065, 0.13706645250761, 0.0353051958283927,
0.0675089177643189, 0.0281994399606395, 0.0308050141564294, 0.0468025179688982,
0.0433720892664746, 0.0435291452924019, 0.058347095833611, 0.0357374228816215,
0.0681285144965192, 0.0369496928239823), V2 = c(0.0324993125421092,
0.0829902459586018, 0.0313349988480043, 0.0231643111737331, 0.0215266640657336,
0.139950312497446, 0.0468625717360324, 0.0628881811297733, 0.0630012537867713,
0.0761195396634594, 0.110559355829023, 0.0362247903793203, 0.072377746167741,
0.0715725363320271, 0.0547033287830611, 0.0848661283961139, 0.0706141645943618,
0.00893898536731884, 0.029939876404686, 0.0362823949369481, 0.0163019762720283,
0.0260952814925765, 0.0308446486535392, 0.0474220518309602, 0.107438262031909,
0.0418532416880588, 0.0382999231041663, 0.091012219675286, 0.0648204656488163,
0.0839842625832919, 0.116970879309187, 0.0307685506815593, 0.062127229044436,
0.0287051745494615, 0.0305664367028215, 0.0279122231108484, 0.0518771359629128,
0.0497960697478479, 0.0521941533943811, 0.0369405923159022, 0.0601799820549475,
0.0379999332499946)), class = "data.frame", row.names = c(NA,
-42L))

# Second data frame
b <- structure(list(ReplicateID = structure(1:42, .Label = c("A_1",
"A_2", "A_3", "A_4", "A_5", "A_H", "C_1", "C_2",
"C_3", "C_4", "C_5", "C_H", "B_1", "B_2",
"B_3", "B_4", "B_5", "B_H", "D_1", "D_2",
"D_3", "D_4", "D_5", "D_H", "E_1", "E_2",
"E_3", "E_4", "E_5", "E_H", "F_1", "F_2",
"F_3", "F_4", "F_5", "F_H", "G_1",
"G_2", "G_3", "G_4", "G_5", "G_H"
), class = "factor"), Population = structure(c(1L, 1L, 1L, 1L,
1L, 8L, 3L, 3L, 3L, 3L, 3L, 8L, 2L, 2L, 2L, 2L, 2L, 8L, 4L, 4L,
4L, 4L, 4L, 8L, 5L, 5L, 5L, 5L, 5L, 8L, 6L, 6L, 6L, 6L, 6L, 8L,
7L, 7L, 7L, 7L, 7L, 8L), .Label = c("A", "B", "C", "D",
"E", "F", "G", "H"), class = "factor"), V1 = c(0.103127706373277,
0.0366480195400783, 0.137719674247557, 0.139422866363668, 0.150232530061099,
0.106126088744782, 0.0978214851227684, 0.0518466234943674, 0.0481753136100123,
0.0414562358327376, 0.0288074306893453, 0.185898723313358, 0.0775931366542643,
0.0466631045245785, 0.0587804407949194, 0.0463179008048913, 0.0451060393447877,
0.494658430595772, 0.112172251203402, 0.102250430531479, 0.212156228432713,
0.116950406750474, 0.167428567949051, 0.129243889179497, 0.0308081890019437,
0.0967806301757193, 0.0742627146969085, 0.0572442473648204, 0.0582240864307848,
0.0923892788675954, 0.0197716293862682, 0.0960150603597121, 0.0454382141524711,
0.159737374600057, 0.107136131028671, 0.15482878114399, 0.0998108323167759,
0.104423243666307, 0.0565637412771132, 0.245455019339755, 0.0560234421862343,
0.252932938767422), V2 = c(0.11028631408855, 0.031793163123734,
0.160367825526847, 0.129251068749367, 0.153821245504388, 0.0952719082608454,
0.0992511212764156, 0.0582462967953366, 0.0590063196447002, 0.0462578567360106,
0.0258426149077907, 0.189078114649443, 0.0350670088580497, 0.0431229618406762,
0.0516396135281783, 0.0399432679743854, 0.0430439656869387, 0.476040509529608,
0.130981456785079, 0.108939037150918, 0.237760681746215, 0.127312638384749,
0.134525039982418, 0.137825072093384, 0.0312337954330455, 0.104336296328514,
0.0856056163787312, 0.0375001213623101, 0.0619567325971755, 0.104447304229906,
0.0220907916861529, 0.110171926638918, 0.0435027079796142, 0.189331111689673,
0.115602905858747, 0.175620579014666, 0.0931222933712027, 0.085454919845759,
0.0662949274473576, 0.231371724468241, 0.0563282073660222, 0.223015556775686
)), class = "data.frame", row.names = c(NA, -42L))

# Plots
p1 <- ggplot(a, aes(x=V1, y=V2)) +
        geom_point(aes(color=Population)) +
        stat_smooth(method=lm, se=FALSE) +
        theme(legend.position="none") +
        stat_cor(method="pearson", label.x.npc="left", label.y.npc="top", hjust=0)

p2 <- ggplot(b, aes(x=V1, y=V2)) +
        geom_point(aes(color=Population)) +
        stat_smooth(method=lm, se=FALSE) +
        theme(legend.position="none") +
        stat_cor(method="pearson", label.x.npc="left", label.y.npc="top", hjust=0)

# Panel
ggarrange(p1, p2, ncol=2, nrow=1, align="hv")
```

**Comment by ShanSabri** [comment]
I'm also having the same issue with the labels not aligning across facets.

**Comment by e-leib** [comment]
Has this been resolved?

I'm also having an issue where the labels are not aligning across facets. It looks like they are "climbing" down some stairs in terms of their positioning, and nothing I am doing is lining the up. They still look like they are climbing.

**Comment by Jeffrothschild** [comment]
I have struggled with the same issue and found that inherit.aes = FALSE solves it. The problem seems to be when you have colors mapped to shape/fill/etc.

**Comment by asatornal** [comment]
Could you please leave an example on how to use  inherit.aes = FALSE 
Thanks a lot!

**Comment by Jeffrothschild** [comment]
Sure @asatornal 


```
# not what we want
mtcars %>% as_tibble() %>% 
  mutate(cyl = factor(cyl)) %>% 
  ggplot(aes(mpg, hp, fill= cyl, color = cyl))+
  geom_point() +
  geom_smooth(se = F, method = "lm", show.legend = F) +
  ggpubr::stat_cor(r.accuracy = 0.01, cor.coef.name = "r", aes(label = paste(..r.label..)), color = "black",
                   label.y.npc="top", label.x.npc = "left", inherit.aes = TRUE) +
  facet_wrap(~cyl) 


# what we want
mtcars %>% as_tibble() %>% 
  mutate(cyl = factor(cyl)) %>% 
ggplot(aes(mpg, hp, fill= cyl, color = cyl))+
  geom_point() +
  geom_smooth(se = F, method = "lm", show.legend = F) +
  facet_wrap(~cyl) +
  ggpubr::stat_cor(r.accuracy = 0.01, cor.coef.name = "r", aes(x = mpg, y = hp, label = paste(..r.label..)), inherit.aes = FALSE) 
  
```

**Comment by asatornal** [comment]
Thank you for your quick answer! I cant get it to work when I use default method (LOESS and not "lm") together with scales = "free_y". Any idea why?

mtcars %>% as_tibble() %>% 
  mutate(cyl = factor(cyl)) %>% 
  ggplot(aes(mpg, hp, fill= cyl, color = cyl))+
  geom_point() +
  geom_smooth(se = F, show.legend = F) +
  ggpubr::stat_cor(r.accuracy = 0.01, cor.coef.name = "r", aes(x = mpg, y = hp, label = paste(..r.label..)), inherit.aes = FALSE) +
  facet_wrap(~cyl, scales = "free_y")

**Comment by Jeffrothschild** [comment]
@asatornal sorry just seeing this, but I have no idea ¯\_(ツ)_/¯

**Comment by caparks2** [comment]
The solution given above to set `inherit.aes = FALSE` didn't work for me.  It appears the inconsistent placement of the label is due to the label being placed at the maximum y axis data value, rather than the top of the y axis in normalized npc units. However, using `label.y.npc = 1` didn't change anything. The labels were still inconsistent. I worked around the issue by using `label.y`, manually setting this values to `max(y) * 1.025` to set the placement of the label in terms of y axis data units, plus a multiplier to lift it slightly above the maximum y axis value so as to not overlap with points.

``` r
library(ggplot2)
library(ggpubr)

# minimal example data set
d <- data.frame(
  x = c(1, 2, 3, 1, 1, 2, 3, 1),
  y = c(1, 2, 3, 3, 1, 2, 2, 2),
  z = c("A", "A", "A", "A", "B", "B", "B", "B")
)

# reproducing the inconsistent y label positions
d |> ggplot(aes(x, y)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  stat_cor() +
  facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/G4oEc3Z.png)

``` r

# setting `inherit.aes = FALSE` doesn't work
d |> ggplot(aes(x, y)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  stat_cor(aes(x, y), inherit.aes = FALSE) +
  facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/ph7zjAH.png)

``` r

# setting `label.y.npc = 1` doesn't work
d |> ggplot(aes(x, y)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  stat_cor(label.y.npc = 1) +
  facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/ZYm30cA.png)

``` r

# manually fixing the y label positions
d |> ggplot(aes(x, y)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  stat_cor(label.y = max(d$y) * 1.025) +
  facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/cjsYGGO.png)

**Comment by QuintenSand** [comment]
This problem has also been asked on StackOverflow [here](https://stackoverflow.com/questions/76908315/ggpubrstat-cor-label-npc-doesnt-take-into-account-geom-smooth-output/76914879#76914879). Here is a workaround by using `ggplot_build` to manually change the y-coordinates of the labels. Here is some reproducible code I created of my answer in the SO question: 

``` r
library(ggplot2)
library(ggpubr)
data('iris')
p <- ggplot(data = iris, aes(x = Petal.Length, y = Sepal.Length, color = Species))+
  geom_point()+
  geom_smooth(method = 'lm')+
  facet_wrap('Species', scales = 'free')+
  stat_cor(aes(color = NULL),
           size = 4, family = 'serif', show.legend = F, p.accuracy = 0.001)
p
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/XWFPedk.png)

``` r

q <- ggplot_build(p)
#> `geom_smooth()` using formula = 'y ~ x'
# Change coordinates
q$data[[3]]$y = c(5.8, 7, 8.2)

# Turn back
q <- ggplot_gtable(q)

# Plot again
plot(q)
```

![](https://i.imgur.com/xGEEsTa.png)

<sup>Created on 2023-08-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by aphalo** [comment]
There is no need to edit the ggplot as a workaround for what looks like a long-standing bug if one uses `stat_correlation()` from 'ggpmisc' instead of `stat_cor()` from 'ggpubr'. Some functions were copied long ago from 'ggpmisc' into 'ggpubr' and renamed, others like `stat_cor()` only reused some of the code (original authorship was never acknowledged in 'ggpubr'). Maintenance and development of 'ggpmisc' has been more active than that of 'ggpubr' as I have fixed bugs as soon as reported. You may want to give it a try. Example code is shown in my answer to the [same question at StackOverflow](https://stackoverflow.com/questions/76908315/ggpubrstat-cor-label-npc-doesnt-take-into-account-geom-smooth-output/76914879#76914879)

**Comment by wbvguo** [comment]
**TL;DR**: Here are the solutions I found. Either one solves the problem for me
- aesthetic setting: `stat_cor(color = "black")`
- aesthetic mapping: `stat_cor(mapping = aes(color = ${factor_var}, group=1))`

replace `${factor_var}` to the color variable of your design

</br>

--------

I came across the same phenomenon when using `stat_cor()` with `facet_wrap()`. The above threads suggest `inherit.aes = FALSE` or manually setting the `label.y` sometimes works, which makes me speculate it has something to do with the variable types (specifically factor). The following are some simple examples

```r
# the position is consistent when the color variable is not factor
mtcars %>% 
  ggplot(aes(x = mpg, y=hp)) + 
  facet_wrap(~cyl, scales = "free_y") + 
  stat_cor(aes(color=cyl)) 
```
![image](https://github.com/kassambara/ggpubr/assets/31968837/57592c76-45db-4326-8926-f8e16ac509e6)


```r
# the position is shifted when the variable is factor
mtcars %>% 
  ggplot(aes(x = mpg, y=hp)) + 
  facet_wrap(~cyl, scales = "free_y") + 
  stat_cor(aes(color=factor(cyl))) 
```
![image](https://github.com/kassambara/ggpubr/assets/31968837/fd77ebe5-27ee-4c6f-9649-01fcc433ceab)


Therefore we can solve the inconsistency issue by override the aesthetic mapping, e.g. by aesthetic setting
```r
mtcars %>% 
  ggplot(aes(x = mpg, y=hp)) + 
  facet_wrap(~cyl, scales = "free_y") + 
  stat_cor(color = "black") 
```
![image](https://github.com/kassambara/ggpubr/assets/31968837/00131413-047e-40e3-9619-2bec82d3d8a9)

If you still want to preserve the color, then try to set the `group` variable to be the same in aesthetic mapping
```r
mtcars %>% 
    ggplot(aes(x = mpg, y=hp)) + 
    facet_wrap(~cyl, scales = "free_y") + 
    stat_cor(aes(color = factor(cyl), group=1))  
```
![image](https://github.com/kassambara/ggpubr/assets/31968837/69a94394-f857-4e9f-81f8-63aab4cb5a22)


Hope it helps.

Best,
Wenbin

**Comment by aphalo** [comment]
@wbvguo Using `group=1` is a nice and useful trick!  

`stat_correlation()` from 'ggpmisc' has parameters called `vstep` and `hstep` to control the position shift and in addition uses the range extracted from the scale instead of the range of the data. As the code in 'ggpubr' is based on an old version of another stat from 'ggpmisc', which I have later fixed, I can explain why the positions shift when there is a grouping created by mapping a factor to an aesthetic that is also used to create facets: those labels are positioned as if they were in the same panel, and positions shifted to avoid an, in this special case, inexistent overlap. As far as I know, it is impossible or at least difficult to work out which levels of a factor are present or abscent in each panel of a plot using facets. The solution implemented in 'ggpmisc' is given by setting `vstep = 0` to disable the shift. This parameter can also be used to adjust the spacing. Alternatively, `vstep` can be passed a longer vector as argument for control of the position of individual labels. (`hstep` controls the horizontal shift.)

There is, however, a drawback to using the range extracted from the scale: the range is available only if something else is plotted. So in this example I added `geom_point()`. I expect to be able to solve this now that 'ggplot2' 3.5.0 supports NPC coordinates nativelly, eliminating the need to compute positions based on the range of the plotting area.

``` r
library(ggpmisc)

ggplot(mtcars, aes(x = mpg, y=hp)) + 
  geom_point() +
  facet_wrap(~cyl, scales = "free_y") + 
  stat_correlation(aes(color = factor(cyl)), vstep = 0)
```

![](https://i.imgur.com/Ak4X0cr.png)

<sup>Created on 2024-03-26 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by wbvguo** [comment]
Hi @aphalo, thanks for your explanation!

Yes, indeed I also noticed the shortcoming of using the data scale to determine the label position in `stat_cor()`. In the following example, the labels positions are still not consistent when the data ranges are different across facets.

```r
mtcars %>% 
    ggplot(aes(x = mpg, y=hp, color=factor(cyl))) + 
    facet_wrap(~cyl, scales = "free") + 
    stat_cor(aes(group =1)) +
    geom_point() +
    geom_smooth(method = "lm", se = TRUE)
```

![image](https://github.com/kassambara/ggpubr/assets/31968837/87a2cb96-ccf9-4d50-b731-b9e3481802ac)

**Comment by kassambara** [comment]
This is now improved in the development version (install with `remotes::install_github("kassambara/ggpubr")`). `stat_cor()` and `stat_regline_equation()` gain a `label.y.step` argument that fixes the most common case in this thread (labels "climbing stairs" across facets). The original data-range/`geom_smooth` case is separate and still needs a workaround, shown below.

Thanks all for the detailed reports. There are actually **two** separate causes
of misaligned `stat_cor()` labels in this thread; here is each with a fix
(using the current dev version).

``` r
suppressPackageStartupMessages({ library(ggpubr); library(dplyr) })
#> Warning: package 'ggplot2' was built under R version 4.5.2
```

## Cause 1 - labels “climbing stairs” across facets (the most common case here)

When a factor is mapped to an aesthetic (e.g. `color`) that also defines the
facets, each panel holds a single group, but the per-group vertical shift is
applied using the *global* group index, so labels step down panel by panel.

``` r
mt <- mtcars %>% mutate(cyl = factor(cyl))
base <- ggplot(mt, aes(mpg, hp)) + geom_point() + facet_wrap(~cyl, scales = "free_y")
```

Before - `stat_cor(aes(color = cyl))` staggers the labels:

``` r
base + stat_cor(aes(color = cyl))
```

![](https://i.imgur.com/MaqgV6B.png)

Fix - the new `label.y.step = 0` disables the per-group shift so labels align
across panels (the explicit equivalent of the `aes(group = 1)` trick):

``` r
base + stat_cor(aes(color = cyl), label.y.step = 0)
```

![](https://i.imgur.com/KjPEHCT.png)

## Cause 2 - the original report: label anchored to the data range + `geom_smooth`

Here each panel has a single group, but the label is placed relative to the
*data* range while `geom_smooth()` extends each panel by a different amount, so
the same `label.y.npc = "top"` lands at a different height. `label.y.step` does
NOT fix this (it only controls the per-group shift from Cause 1).

``` r
d <- data.frame(
  x = c(1, 2, 3, 1, 1, 2, 3, 1),
  y = c(1, 2, 3, 3, 1, 2, 2, 2),
  z = c("A", "A", "A", "A", "B", "B", "B", "B")
)
base <- ggplot(d, aes(x, y)) + geom_point() + geom_smooth(method = "lm", se = FALSE)
```

Before - with `scales = "free_y"` the two labels sit at different heights:

``` r
base + stat_cor() + facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/GZZgsSP.png)

Fix - make the y range shared, so the labels align. Either use a shared y axis
(`scales = "fixed"`, the facet default)…

``` r
base + stat_cor() + facet_wrap(vars(z))
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/ixjc3fF.png)

…or set an explicit `label.y` (works with `free_y` too, and for separate
plots combined with `ggarrange()` use a common `ylim()`):

``` r
base + stat_cor(label.y = 3.1) + facet_wrap(vars(z), scales = "free_y")
#> `geom_smooth()` using formula = 'y ~ x'
```

![](https://i.imgur.com/j2ZVqMd.png)

(ggpubr’s correlation label-positioning logic was adapted from the ‘ggpmisc’
package by Pedro J. Aphalo; this is now credited via `@seealso` in `?stat_cor`,
and `label.y.step` was inspired by ggpmisc’s `vstep`.)

<sup>Created on 2026-07-02 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

---

I'm keeping this issue **open** for the eventual switch to native panel-NPC label positioning (ggplot2 >= 3.5.0 supports NPC natively), which would remove the need for the Cause 2 workarounds entirely.

**Comment by kassambara** [comment]
The original "Cause 2" (labels drifting because they are anchored to the *data* range, which each panel/plot extends differently) is now addressed in the development version. `stat_cor()` and `stat_regline_equation()` gain a `label.anchor` argument; with `label.anchor = "panel"` the label is placed at the true panel-relative position, so it stays aligned across facets or plots whose axis ranges differ.

```r
# remotes::install_github("kassambara/ggpubr")
library(ggpubr)

d <- data.frame(x = c(1,2,3,1,1,2,3,1), y = c(1,2,3,3,1,2,2,2),
                z = c("A","A","A","A","B","B","B","B"))

ggplot(d, aes(x, y)) + geom_point() + geom_smooth(method = "lm", se = FALSE) +
  stat_cor(label.anchor = "panel") +
  facet_wrap(vars(z), scales = "free_y")
```

![default vs label.anchor = "panel"](https://i.imgur.com/RzDkvPg.png)

Top: the default (`label.anchor = "data"`) places each label at that panel's data maximum, so with `scales = "free_y"` + `geom_smooth()` they sit at different heights. Bottom: `label.anchor = "panel"` places them at the true panel top, so they align. It also works for separate plots combined with `ggarrange()` (use it in each plot).

Together with `label.y.step` (which handles the per-group "stair-stepping" reported later in the thread), both causes of misaligned `stat_cor()` labels now have a solution. The default `label.anchor = "data"` is unchanged, so existing plots are unaffected. I'll leave this open in case it's worth making panel-anchoring the default in a future release.

**Comment by kassambara** [comment]
Both causes of misaligned `stat_cor()`/`stat_regline_equation()` labels raised in this thread now have a solution in the development version:

- **Per-group "stair-stepping" across facets** → `label.y.step = 0` (or a custom step).
- **Labels anchored to the data range drifting across panels/plots** (the original report) → `label.anchor = "panel"`, which places the label at the true panel-relative position so it stays aligned regardless of `scales = "free_y"`, `geom_smooth()` extension, or `ggarrange()` of plots with different ranges.

Closing as resolved. Making panel-anchoring the default is a separate consideration for a future release and can be tracked on its own if pursued. Thanks all for the detailed reports and reproducible examples.
